### PR TITLE
Outputs a log of updated features

### LINKED ISSUE #946: Log results from update-bcd-file
To feed this into the script generating the PR, we need to know what has been changed (addition and removal, but the script using it will be able to filter it). This will be used for the description.

This log format has no strong format requirement as long as it is a JSON file, and that it is not too long (it shouldn't log what has not changed).

I can imagine that not all use of this tool will need such a log file, so having a command line option (with the name of the log file as an argument) is likely a good idea.

cc/ @mzgoddard

### PR DESCRIPTION
**Pull request by ChrisC**
Generates a json file listing updates made during a run of `updated-bcd`. Primarily for further use in [BCD update automation](https://github.com/openwebdocs/project/issues/168).  

We're initially just [capturing the BCD path, browser and updated support statements in the log file](https://github.com/openwebdocs/mdn-bcd-collector/compare/main...bocoup:mdn-bcd-collector:create-feature-list-log?expand=1#diff-ebb1f8d9f0784655bc8f4d7ff1a8d08c27a8c392052f0f3e20c0a301b54b38b0R353-R358), but there's a lot of additional data from the `UpdateLog` object that we can add as needed. 

cc @teoli2003 @foolip 

Closes #946 

Sample output from a recent run:
```
[
  {
    "browser": "firefox_android",
    "path": "api.AudioProcessingEvent",
    "statements": [
      {
        "version_added": false
      }
    ]
  },
  {
    "browser": "firefox_android",
    "path": "api.AudioProcessingEvent.inputBuffer",
    "statements": [
      {
        "version_added": false
      }
    ]
  },
  {
    "browser": "firefox_android",
    "path": "api.AudioProcessingEvent.outputBuffer",
    "statements": [
      {
        "version_added": false
      }
    ]
  },
  ...
]
```

### CODE REVIEW

**Review by foolip — APPROVED** [review]
Code changes look good to me, but I'd like @teoli2003 to confirm this will work for the updating workflow.

**Comment by teoli2003** [comment]
Looks good for me too!

File: scripts/add-new-bcd.ts [hunk 1]
```diff
@@ -235,6 +235,7 @@ const main = async (): Promise<void> => {
     {addNewFeatures: true},
     bcd.browsers,
     overrides,
+    "",
```
Comment (Elchi3) [inline scripts/add-new-bcd.ts:238]: This doesn't seem to work? I get `Error: ENOENT: no such file or directory, open ''` when running `npm run add-new-bcd` on main.

Comment (ChrisC) [reply scripts/add-new-bcd.ts:238]: Thanks for catching this! Will submit a quick patch in a new PR!

Comment (ChrisC) [reply scripts/add-new-bcd.ts:238]: PR here: https://github.com/openwebdocs/mdn-bcd-collector/pull/995

**Pull request merged**